test/date-time/date-time-math.test.ts

```diff
@@ -0,0 +1,281 @@
+            checkIsCorrect(dateTime, Duration.fromMinutes(61), "61 minutes", "2024-01-01 08:59");
+        });
+
+        suite("Subtract hours", () =>
+        {
+            checkIsCorrect(dateTime, Duration.fromHours(0), "0 hours", dateTime.value);
+            checkIsCorrect(dateTime, Duration.fromHours(1), "1 hour", "2024-01-01 09:00");
+            checkIsCorrect(dateTime, Duration.fromHours(24), "24 hours", "2023-12-31 10:00");
+            checkIsCorrect(dateTime, Duration.fromHours(25), "25 hours", "2023-12-31 09:00");
+        });
+
+        suite("Subtract days", () =>
+        {
+            checkIsCorrect(dateTime, Duration.fromDays(0), "0 days", dateTime.value);
+            checkIsCorrect(dateTime, Duration.fromDays(1), "1 day", "2023-12-31 10:00");
+            checkIsCorrect(dateTime, Duration.fromDays(31), "31 days", "2023-12-01 10:00");
+            checkIsCorrect(dateTime, Duration.fromDays(365), "365 days", "2023-01-01 10:00");
+            checkIsCorrect(dateTimeLeapYear, Duration.fromDays(366), "366 days", "2024-01-01 10:00");
+        });
+
+        suite("Subtract weeks", () =>
+        {
+            checkIsCorrect(dateTime, Duration.fromWeeks(0), "0 weeks", dateTime.value);
+            checkIsCorrect(dateTime, Duration.fromWeeks(1), "1 week", "2023-12-25 10:00");
+            checkIsCorrect(dateTime, Duration.fromWeeks(365 / 7), `365 / 7 weeks`, "2023-01-01 10:00");
+            checkIsCorrect(dateTimeLeapYear, Duration.fromWeeks(366 / 7), `366 / 7 weeks`, "2024-01-01 10:00");
+        });
+    });
+
+
+    suite("Add days", () =>
+    {
+        const dateTime = new DateTime({ value: "2024-01-01 10:00", zone: "utc" });
+        const dateTimeNonLeapYear = new DateTime({ value: "2025-01-01 10:00", zone: "utc" });
+
+        suite("Add days", () =>
+        {
+            function checkIsCorrect(dateTime: DateTime, days: number, expectedValue: string): void
+            {
+                given(dateTime, "dateTime").ensureHasValue().ensureIsType(DateTime);
+                given(days, "days").ensureHasValue().ensureIsNumber();
+                given(expectedValue, "expectedValue").ensureHasValue().ensureIsString()
+                    .ensure(t => DateTime.validateDateTimeFormat(t));
+
+                test(`Given a DateTime (${dateTime.toString()}), and number of days ${days}
+                when number of days is added to the dateTime
+                then it should return value of ${expectedValue}`,
+                    () =>
+                    {
+                        Assert.strictEqual(dateTime.addDays(days).value, expectedValue);
+                    }
+                );
+            }
+
+            checkIsCorrect(dateTime, 0, dateTime.value);
+            checkIsCorrect(dateTime, 1, "2024-01-02 10:00");
+            checkIsCorrect(dateTime, 31, "2024-02-01 10:00");
+            checkIsCorrect(dateTime, 366, "2025-01-01 10:00");
```

Comment:
Can you check what happens when you add a day and that's the day DST start/ends does the time change?